src/main/java/com/iexec/common/tee/TeeEnclaveConfiguration.java

```diff
@@ -30,7 +31,11 @@
 @Builder
 public class TeeEnclaveConfiguration {
 
+    /** @deprecated */
+    @Deprecated(forRemoval = false)
+    @JsonInclude(JsonInclude.Include.NON_NULL)
     private TeeEnclaveProvider provider;
+    private TeeFramework framework;
```

Comment:
```suggestion
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private TeeFramework framework;
```
?

Reply:
I am not sure about this. `TeeFramework` should never be null here.
Unless I'm missing something, a `TeeEnclaveConfiguration` not backed on a TEE framework has no meaning.
I added the annotation on `provider` to cleanup the JSON when core and worker will shift from `provider` to `framework` field.

Reply:
Your are right, I was think about `@NonNull` !